controlplane/kubeadm/controllers/scale.go

```diff
@@ -113,13 +107,6 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 		return ctrl.Result{}, errors.New("failed to pick control plane Machine to delete")
 	}
 
-	// Ensure etcd is healthy prior to attempting to remove the member
-	if err := r.managementCluster.TargetClusterEtcdIsHealthy(ctx, util.ObjectKey(cluster), kcp.Name); err != nil {
-		logger.V(2).Info("Waiting for control plane to pass etcd health check before removing a control plane machine", "cause", err)
-		r.recorder.Eventf(kcp, corev1.EventTypeWarning, "ControlPlaneUnhealthy",
-			"Waiting for control plane to pass etcd health check before removing a control plane machine: %v", err)
-		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
-	}
 	// If etcd leadership is on machine that is about to be deleted, move it to the newest member available.
 	etcdLeaderCandidate := ownedMachines.Newest()
 	if err := workloadCluster.ForwardEtcdLeadership(ctx, machineToDelete, etcdLeaderCandidate); err != nil {
```

Comment:
I'm wondering if it would make sense to consolidate the etcd membership removal to just the health reconciliation, rather than duplicating it here as well? 

Similar with the kubeadm configmap modification below as well.

Reply:
Health reconciliation removes etcds only when machine is deleted. In ForwardEtcdLeadership case, machine is still healthy. So consolidating them means waiting until the next health reconciliation. Am I missing something?

Reply:
Yeah, it would delay the removal until the next reconciliation. It's probably best to keep it duplicated for now to avoid disruption of the leader.